main/src/ca/uwaterloo/flix/language/phase/unification/SetFormula.scala

```diff
@@ -221,6 +299,63 @@ object SetFormula {
     }
   }
 
+  /**
+    * Extracts the index type (kind CaseSet) of a restrictable enum along
+    * with its enumSym.
+    * {{{
+    * findIndexOfEnum(E[index: CaseSet(Color), Int32, String]) = Some((index, Color))
+    * findIndexOfEnum(Int32) = None
+    * findIndexOfEnum(Int32 -> E[index]) = None
+    * }}}
+    */
+  @tailrec
+  private def findIndexOfEnum(tpe: Type): Option[(Type, Symbol.RestrictableEnumSym)] = tpe match {
```

Comment:
Question: Why not do a kind check and then `tpe.typeArguments.head`?

Reply:
The kind of `tpe` is `Kind.Star`, but I could find `typeArguments.head` and then check if that's a case set?